e2etests/web/regular_integration_tests/lib/screenshot_support.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:golden_comparator/goldens.dart';
+import 'package:integration_test/integration_test_driver_extended.dart' as test;
+
+import 'package:golden_comparator/image_compare.dart';
+
+import 'package:image/image.dart';
+
+/// Tolerable pixel difference ratio between the goldens and the screenshots.
+///
+/// We are allowing a higher difference rate compared to the unit tests (where
+/// this rate is set to 0.28), since during the end to end tests there are
+/// more components on the screen which are not related to the functinality
+/// under test ex: a blinking cursor.
+const double kMaxDiffRateFailure = 0.5 / 100; // 0.5%
+
+/// Used for calling `integration_test` package.
+///
+/// Compared to other similar classes which only included the following call:
+/// ```
+/// Future<void> main() async => test.integrationDriver();
+/// ```
+///
+/// this method is able to take screenshot.
+///
+/// It provides an `onScreenshot` callback to the `integrationDriver` method.
+/// It also includes options for updating the golden files.
+Future<void> main() async {
```

Comment:
I'm not sure if the name `main` is helpful since we don't use this file as an entrypoint. How about `runTestWithScreenshots`? This will also remove the need to import this library with the `with_screenshot` prefix.

Reply:
done